tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala

```diff
@@ -495,6 +524,31 @@ class TestQuintEx extends AnyFunSuite {
     assert(convert(Q.app("with", rec, Q.s, Q._42)(typ)) == """[["s" ↦ 1, "t" ↦ 2] EXCEPT ![<<"s">>] = 42]""")
   }
 
+  test("`with` operator conversion preserves row-typing") {
+    val typ = QuintRecordT.ofFieldTypes("a", ("s", QuintIntT()), ("t", QuintIntT()))
+    val rec = Q.app("Rec", Q.s, Q._1, Q.t, Q._2)(typ)
+    val exp = Q.quint.exToTla(rec).get
+    val expectedTlaType = RecRowT1(RowT1(VarT1("a"), ("s", IntT1), ("t", IntT1)))
+
+    assert(Quint.typeToTlaType(typ) == expectedTlaType)
+    assert(exp.typeTag == Typed(expectedTlaType))
+  }
```

Comment:
This seems off place, I think it's an old version of the test added above? It doesn't seem to be testing the `with` operator.

Reply:
Oh yeah. That was an oversight, this got covered in the best below. Thanks!